pkg/api/quota_test.go

```diff
@@ -28,127 +33,156 @@ var testOrgQuota = setting.OrgQuota{
 	AlertRule:  10,
 }
 
-// setupDBAndSettingsForAccessControlQuotaTests stores users and create two orgs
-func setupDBAndSettingsForAccessControlQuotaTests(t *testing.T, sc accessControlScenarioContext) {
-	t.Helper()
-
-	// Create two orgs with the context user
-	setupOrgsDBForAccessControlTests(t, sc.db, sc, 2)
-}
-
 func TestAPIEndpoint_GetCurrentOrgQuotas_LegacyAccessControl(t *testing.T) {
 	cfg := setting.NewCfg()
 	cfg.Quota.Enabled = true
 	cfg.RBACEnabled = false
-	sc := setupHTTPServerWithCfg(t, true, cfg)
-	setInitCtxSignedInViewer(sc.initCtx)
-
-	setupDBAndSettingsForAccessControlQuotaTests(t, sc)
+	server := SetupAPITestServer(t, func(hs *HTTPServer) {
+		hs.Cfg = cfg
+	})
 
 	t.Run("Viewer can view CurrentOrgQuotas", func(t *testing.T) {
-		response := callAPI(sc.server, http.MethodGet, getCurrentOrgQuotasURL, nil, t)
-		assert.Equal(t, http.StatusOK, response.Code)
+		req := webtest.RequestWithSignedInUser(server.NewGetRequest(getCurrentOrgQuotasURL), &user.SignedInUser{OrgID: 1, OrgRole: org.RoleViewer})
+		res, err := server.Send(req)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusOK, res.StatusCode)
+		require.NoError(t, res.Body.Close())
 	})
 
-	sc.initCtx.IsSignedIn = false
 	t.Run("Unsigned user cannot view CurrentOrgQuotas", func(t *testing.T) {
-		response := callAPI(sc.server, http.MethodGet, getCurrentOrgQuotasURL, nil, t)
-		assert.Equal(t, http.StatusUnauthorized, response.Code)
+		req := server.NewGetRequest(getCurrentOrgQuotasURL)
+		res, err := server.Send(req)
+		require.NoError(t, err)
+		assert.Equal(t, http.StatusOK, res.StatusCode)
```

Comment:
Shouldn't this be `http.StatusUnauthorized` instead?

Reply:
You're very right! Thanks for pointing it out 🕵️